operator/pkg/values/translation/translation_test.go

```diff
@@ -180,3 +181,382 @@ func TestGetTLSConfigValues(t *testing.T) {
 		})
 	}
 }
+
+func TestSetScannerComponentDisabled(t *testing.T) {
+	tests := map[string]struct {
+		scannerComponent platform.ScannerComponentPolicy
+		want             chartutil.Values
+		wantErr          bool
+	}{
+		"Disabled": {
+			scannerComponent: platform.ScannerComponentDisabled,
+			want: chartutil.Values{
+				"disable": true,
+			},
+		},
+		"Enabled": {
+			scannerComponent: platform.ScannerComponentEnabled,
+			want: chartutil.Values{
+				"disable": false,
+			},
+		},
+		"Invalid": {
+			scannerComponent: "invalid",
+			want:             chartutil.Values{},
+			wantErr:          true,
+		},
+	}
+
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			vb := NewValuesBuilder()
+			scannerComponent := tt.scannerComponent
+			SetScannerComponentDisabledValue(&vb, &scannerComponent)
+			values, err := vb.Build()
+			if tt.wantErr {
+				require.NotNil(t, err)
+				return
+			}
+
+			assert.NoError(t, err)
+			assert.Equal(t, tt.want, values)
+		})
+	}
+}
+
+func TestSetScannerV4ComponentValues(t *testing.T) {
+	// using local copies to be able to create references to those constants
+	autoscalingEnabled := platform.ScannerAutoScalingEnabled
+	autoscalingDisabled := platform.ScannerAutoScalingDisabled
+	var autoscalingInvalid platform.AutoScalingPolicy = "invalid"
```

Comment:
up to you but I think you can also do `autoscalingInvalid := platform.AutoScalingPolicy("invalid")`